### PR TITLE
Add ToSql / FromSql for IpInet and IpCidr from cidr crate

### PR DESCRIPTION
**Pull request by timando**

### CODE REVIEW

File: postgres-types/src/cidr_02.rs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    accepts!(CIDR);
+    to_sql_checked!();
+}
+
+impl<'a> FromSql<'a> for IpInet {
+    fn from_sql(_: &Type, raw: &[u8]) -> Result<Self, Box<dyn Error + Sync + Send>> {
+        let inet = types::inet_from_sql(raw)?;
+        Ok(IpInet::new(inet.addr(), inet.netmask())
+            .expect("postgres enforces maximum length of netmask"))
```
Comment (sfackler) [inline postgres-types/src/cidr_02.rs:32]: Same

File: postgres-types/src/cidr_02.rs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+use postgres_protocol::types;
+use std::error::Error;
+
+use crate::{FromSql, IsNull, ToSql, Type};
+
+impl<'a> FromSql<'a> for IpCidr {
+    fn from_sql(_: &Type, raw: &[u8]) -> Result<Self, Box<dyn Error + Sync + Send>> {
+        let inet = types::inet_from_sql(raw)?;
+        Ok(IpCidr::new(inet.addr(), inet.netmask())
+            .expect("postgres cidr type has zeroed host portion"))
```
[the remark this answers is not in the document]
Comment (timando) [reply postgres-types/src/cidr_02.rs:12]: the type only accepts postgres `cidr` type which postgres guarantees to be a valid `cidr` (i.e. mask length within the limits for the address type and zeroed host portion) which is the same as the requirements to create an `IpCidr`.

File: postgres-types/src/cidr_02.rs (hunk 1 above)
Comment (timando) [reply postgres-types/src/cidr_02.rs:32]: Similar, postgres guarantees that the `inet` type has a mask length corresponding to the address type (<= 32 for IPv4, <=128 for IPv6)

File: postgres-types/src/cidr_02.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (sfackler) [reply postgres-types/src/cidr_02.rs:12]: If you assume the value passed to from_sql is always correct, then there's no need for any of these implementations to return an error. The library should not assume that the input is valid.

**Comment by sfackler** [comment]
Thanks!

**Pull request merged**